test/SafetyLocks.t.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import "forge-std/Test.sol";
+
+import { SafetyLocks } from "../src/contracts/atlas/SafetyLocks.sol";
+import { FastLaneErrorsEvents } from "../src/contracts/types/Emissions.sol";
+
+import "../src/contracts/types/DAppApprovalTypes.sol";
+import "../src/contracts/types/LockTypes.sol";
+
+contract MockSafetyLocks is SafetyLocks {
+    constructor() SafetyLocks(0, address(0), address(0)) { }
+
+    function initializeEscrowLock(
+        address executionEnvironment,
+        uint256 gasMarker,
+        uint256 userOpValue
+    )
+        external
+        payable
+    {
+        _initializeEscrowLock(executionEnvironment, gasMarker, userOpValue);
+    }
+
+    function checkIfUnlocked() external view {
+        _checkIfUnlocked();
+    }
+
+    function buildEscrowLock(
+        DAppConfig calldata dConfig,
+        address executionEnvironment,
+        uint8 solverOpCount,
+        bool isSimulation
+    )
+        external
+        view
+        returns (EscrowKey memory escrowKey)
+    {
+        return _buildEscrowLock(dConfig, executionEnvironment, solverOpCount, isSimulation);
+    }
+
+    function releaseEscrowLock() external {
+        _releaseEscrowLock();
+    }
+
+    function setLock(address _lock) external {
+        lock = _lock;
+    }
+
+    function setClaims(uint256 _claims) external {
+        claims = _claims;
+    }
+
+    function setWithdrawals(uint256 _withdrawals) external {
+        withdrawals = _withdrawals;
+    }
+
+    function setDeposits(uint256 _deposits) external {
+        deposits = _deposits;
+    }
+}
+
+contract SafetyLocksTest is Test {
+    MockSafetyLocks public safetyLocks;
+    address executionEnvironment = makeAddr("executionEnvironment");
+
+    function setUp() public {
+        safetyLocks = new MockSafetyLocks();
+    }
+
+    function test_initializeEscrowLock() public {
```

Comment:
There is no test here to ensure that `_initializeEscrowLock` calls `_checkIfUnlocked`.

Reply:
Added in https://github.com/FastLane-Labs/atlas/pull/86/commits/5bac7e91c8c91ad76ed5e0333eedd02b9fad4bd7